src/Navi.js

```diff
@@ -0,0 +1,150 @@
+import { request } from "./Api.js";
+import { push } from "./router.js";
+import { setItem } from "./Storage.js";
+import { CheckNew } from "./Error.js";
+
+export default function Navi({ $target, initialState }) {
+  CheckNew(new.target);
+
+  const $navi = document.createElement("div");
+  $navi.className = "Navi";
+
+  $navi.style = "width:95%;";
+
+  this.state = initialState;
+
+  this.setState = (nextState) => {
+    this.state = nextState;
+    this.render();
+  };
+
+  // 찾고 싶은 문서의 id를 받아온다.
+  // DFS로 해당 id의 문서를 찾는다.
+  this.render = () => {
+    $target.prepend($navi);
+
+    if (this.state.postId === "new") {
+      $navi.innerHTML = `
+        새로운 루트 문서군요!
+      `;
+      return;
+    }
+
+    const { postId } = this.state;
+    fetchPosts(Number(postId));
+  };
+
+  const fetchPosts = async (postId) => {
+    const posts = await request("documents", {
+      metohd: "GET",
+    });
+
+    let postPath = "";
+    let postChildren = "";
+    posts.forEach((item) => {
+      const temp = DFS(item, postId);
+      if (temp !== undefined) {
+        postPath = temp[0];
+        postChildren = temp[1];
+      }
+    });
+
+    pathOpen(postPath);
+
+    $navi.innerHTML = `
+    <p> 파일 경로:          
+      ${
+        postPath.length !== 0
+          ? postPath
+              .map((item) => {
+                return `<span data-id="${item.id}">${item.title}</span>`;
+              })
+              .join(" < ")
+          : "상위 파일 없음"
+      }
+    </p>
+    <p> 하위 파일:
+        ${
+          postChildren.length !== 0
+            ? postChildren
+                .map((item) => {
+                  return `<span data-id="${item.id}">${item.title}</span>`;
+                })
+                .join(" / ")
+            : "하위 파일 없음"
+        }
+    </p>
+    `;
```

Comment:
```suggestion
    $navi.innerHTML = `
      <p> 파일 경로:
        ${renderPath(postPath, "<") || "상위 파일 없음"}
      </p>
      <p> 하위 파일:
        ${renderPath(postChildren, " / ") || "하위 파일 없음"}
      </p>
    `;
```

추상화 수준이 맞지 않습니다. (선언형 마크업에 너무 많은 로직이 들어가 있습니다.)
중복되는 로직도 보이고 함수로 분리하는 방안이 나아 보입니다.